### PR TITLE
feat: Add a loading spinner in the taskbar while the Friends feature is being initialized

### LINKED ISSUE #1748: Improve the way Friends feature fails
**Figma:** https://www.figma.com/file/SxxomcIZI1zyytMZmFJAvS/Social-Module-%5Bwip%5D?node-id=604%3A13360

**Acceptance Criteria**
- [x] Remove Friends connection fail notification
- [x] While the Friends panel is loading, a loading spinner will be showed.
- [x] We won't never contemplate a "failed" status of the friends feature because we maked it be indefinitely retried from kernel.

**NOTE:** Use `feat/SocialBar` as base branch.

## Kernel side
https://github.com/decentraland/kernel/pull/282

### PR DESCRIPTION
**Pull request by sandrade-dcl**
## What does this PR change?
Fix #1748

**Figma:** https://www.figma.com/file/SxxomcIZI1zyytMZmFJAvS/Social-Module-%5Bwip%5D?node-id=604%3A13360

**Acceptance Criteria**
- [x] Remove Friends connection fail notification
- [x] While the Friends panel is loading, a loading spinner will be showed.
- [x] We won't never contemplate a "failed" status of the friends feature because we maked it be indefinitely retried from kernel.

**NOTE:** Use `feat/SocialBar` as base branch.

## Kernel side
https://github.com/decentraland/kernel/pull/282

## How to test the changes?
1. Go to: https://play.decentraland.zone/?renderer-branch=feat/improve-the-way-friends-feature-fails-explorer-side&kernel-branch=feat/improve-the-way-friends-feature-fails-kernel-side
2. Enter the world and wait for friends initialization.
3. Notice that a loading spinner is showed in the taskbar while friends are not initialized.

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Comment by sandrade-dcl** [comment]
**NOTE:** We still need to integrate these final visual assets from Romi:

image

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/FriendsController.cs [hunk 1]
```diff
@@ -67,6 +68,7 @@ public bool ContainsStatus(string friendId, FriendshipStatus status)
     public event Action<string, FriendshipAction> OnUpdateFriendship;
     public event Action<string> OnFriendNotFound;
     public event Action OnInitialized;
+    public event Action OnNotInitialized;
```
[the remark this answers is not in the document]
Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/FriendsController.cs]: Good idea. Done in https://github.com/decentraland/unity-renderer/pull/2228/commits/c61b3d4012c667b1a8635150689086085a611b9b.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/TaskbarHUD/TaskbarHUDController.cs [hunk 2]
```diff
@@ -399,4 +428,24 @@ private void OnFriendsToggleInputPress()
...
+    {
+        view.SetFiendsAsLoading(false);
+        view.SetFriendsAsFailed(true);
+    }
+
+    private void RetryFriendsInitialization()
+    {
+        view.SetFriendsAsFailed(false);
+        view.SetFiendsAsLoading(true);
+        WebInterface.RetryFriendsInitialization();
```
[the remark this answers is not in the document]
Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/TaskbarHUD/TaskbarHUDController.cs:494]: I have tried to do it from IFriendsController but I'm getting a cycle reference in the assembly definitions. To fix that we should separate the current `FriendsController` module in 2 different assemblies: `FriendsController` and `FriendsControllerInterfaces`. I have tried to do it but it is provoking A LOT of changes in the project (more than 70 changed files) and I think we should do it in a separate PR.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/TaskbarHUD/TaskbarHUDController.cs [hunk 3]
```diff
@@ -138,6 +156,9 @@ private void ShowExperiences()
 
     private void ToggleFriendsTrigger_OnTriggered(DCLAction_Trigger action)
     {
+        if (view.isFriendsLoading)
```
Comment (lorux0) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/TaskbarHUD/TaskbarHUDController.cs:159]: you could use `friendsController.IsInitialized`

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/TaskbarHUD/TaskbarHUDController.cs [hunk 4]
```diff
@@ -425,4 +449,25 @@ public void Dispose()
...
+    {
+        bool anyInputFieldIsSelected = EventSystem.current != null &&
+                                       EventSystem.current.currentSelectedGameObject != null &&
+                                       EventSystem.current.currentSelectedGameObject.GetComponent<TMP_InputField>() != null &&
+                                       (!worldChatWindowHud.IsInputFieldFocused || !worldChatWindowHud.IsPreview);
+
+        if (anyInputFieldIsSelected)
+            return;
+
+        Utils.UnlockCursor();
```
Comment (lorux0) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/TaskbarHUD/TaskbarHUDController.cs:463]: you could use `mouseCatcher.UnlockCursor()`

**Review by lorux0 — APPROVED** [review]

**Pull request merged**